### PR TITLE
reduce manual compaction

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
Add a threshold for lock cf's manual compaction. Disable manual compaction for other CFs by default.
ref https://github.com/pingcap/tikv/issues/1663
@siddontang @BusyJay PTAL

### CODE REVIEW

**Comment by siddontang** [comment]
Any test to cover the logic?

**Comment by BusyJay** [comment]
If the compaction for LOCK is disabled, will it be slow when there have been too much write?

File: src/raftstore/store/store.rs [hunk 1]
```diff
@@ -1701,15 +1704,18 @@ impl<T: Transport, C: PdClient> Store<T, C> {
...
-        let task = CompactTask {
-            cf_name: String::from(CF_LOCK),
-            start_key: None,
-            end_key: None,
-        };
-        if let Err(e) = self.compact_worker.schedule(task) {
-            error!("{} failed to schedule compact lock cf task: {:?}",
-                   self.tag,
-                   e);
+        if self.lock_cf_written_bytes > self.cfg.lock_cf_compact_threshold {
```
Comment (siddontang) [inline src/raftstore/store/store.rs]: use a small lock_cf_compact_threshold in test so we can cover the logic.

**Comment by zhangjinpeng1987** [comment]
@disksing PTAL

**Comment by zhangjinpeng1987** [comment]
@overvenus PTAL

**Comment by ngaut** [comment]
LGTM

**Pull request merged**